lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2352,6 +2401,99 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
+    TypeConverter *typeConverter = getTypeConverter();
+
+    
+    if (auto SliceOp = dyn_cast<AtenSliceTensorOp>(op)){
+      AtenSliceTensorOp::Adaptor adaptor(operands);
+      auto input = operands[0];
+      RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+      RankedTensorType resultType =
+          typeConverter->convertType(op->getResult(0).getType()).cast<RankedTensorType>();
+      int64_t rank = resultType.getRank();
+      SmallVector<Value> resultShape;
+      for (int64_t i = 0; i < inputType.getRank(); i++) {
+        auto currentDimSize = rewriter.create<tensor::DimOp>(loc, input, i);
+        resultShape.push_back(currentDimSize);
+      }
+      Value zero, one, end_, start_;
+      zero =  rewriter.create<arith::ConstantIndexOp>(loc, 0);
+      one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+      int64_t dim, start, end, step;
+      if (!matchPattern(SliceOp.dim(), m_TorchConstantInt(&dim)))
+        return op->emitError("unimplemented: dim is not constant");
+      if (!matchPattern(SliceOp.start(), m_TorchConstantInt(&start))) {
+        if (adaptor.start().getType().isa<Torch::NoneType>()) {
+          start = 0;
+          start_ = zero;
+        } else {
+          return op->emitError("unimplemented: start should be int or None");
+        }
+      } else {
+        start_ = rewriter.create<arith::ConstantIndexOp>(loc, start);
+      }
+      if (!matchPattern(SliceOp.end(), m_TorchConstantInt(&end))) {
+        if (adaptor.end().getType().isa<Torch::NoneType>()) {
+          end_ = resultShape[dim];
+          end = -1;
+        } else {
+          return op->emitError("unimplemented: end should be int or None");
+        }
+      } else {
+        end_ =  rewriter.create<arith::ConstantIndexOp>(loc, end);
+      }
+      if (!matchPattern(SliceOp.step(), m_TorchConstantInt(&step)))
+        return op->emitError("unimplemented: step is not constant");
+
+      SmallVector<Value> offsets, strides;
+      strides.resize(rank, one);
+      offsets.resize(rank, zero);
+  //    if (end < 0) 
+  //      end_ = rewriter.create<arith::AddIOp>(loc, end_, resultShape[dim]);
+      Value correctStart = rewriter.create<arith::AddIOp>(loc, start_, resultShape[dim]);
+      Value startIsNegative = rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::slt, start_, zero);
+      start_ = rewriter.create<SelectOp>(loc, startIsNegative, correctStart, start_);
+      Value correctEnd = rewriter.create<arith::AddIOp>(loc, end_, resultShape[dim]);
+      Value endIsNegative = rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::slt, end_, zero);
+      end_ = rewriter.create<SelectOp>(loc, endIsNegative, correctEnd, end_);
+      // len = end - start  end = 5 start = 0 len = 5
+      Value len_ = rewriter.create<arith::SubIOp>(loc, end_, start_);
+      Value step_ = rewriter.create<arith::ConstantIndexOp>(loc, step); 
+      // new_size = (len+ step - 1) / step     (5 + 1 - 1) // 1 == 1
```

Comment:
`floordiv(end - start - 1, step) + 1` can be used for new_size calcuation.

Reply:
oddly, their codebase suggests ceildiv in a comment, but floordiv is correct, thanks